src/components/MyAccount/NewSettings/UsernameForm.test.tsx

```diff
@@ -0,0 +1,154 @@
+import { render, screen, fireEvent, waitFor } from "@testing-library/react"
+import { PatronDataProvider } from "../../../context/PatronDataContext"
+import {
+  emptyPatron,
+  filteredPickupLocations,
+  processedPatron,
+} from "../../../../__test__/fixtures/processedMyAccountData"
+import UsernameForm from "./UsernameForm"
+
+describe("username form", () => {
+  const mockUsernameState = {
+    setUsernameStatus: jest.fn(),
+    setUsernameStatusMessage: jest.fn(),
+  }
+  const accountFetchSpy = jest.fn()
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+    global.fetch = jest.fn().mockResolvedValue({
+      json: async () => {
+        console.log("Updated")
+      },
+      status: 200,
+    } as Response)
+  })
+
+  const component = (
+    <PatronDataProvider
+      testSpy={accountFetchSpy}
+      value={{
+        patron: processedPatron,
+        pickupLocations: filteredPickupLocations,
+      }}
+    >
+      <UsernameForm
+        patron={processedPatron}
+        usernameState={mockUsernameState}
+      />
+    </PatronDataProvider>
+  )
+
+  const noUsernameComponent = (
+    <PatronDataProvider
+      testSpy={accountFetchSpy}
```

Comment:
Unless you want to validate something on this spy, you don't need to provide this value.